### ISSUE TITLE
v195 regression: false positive E0554

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#10420]
## Environment

* **IntelliJ Rust plugin version:** 0.4.195.5367-231-nightly
* **Rust toolchain version:** 1.69.0 (84c898d65 2023-04-16) x86_64-apple-darwin **(!!!)**
* **IDE name and version:** IntelliJ IDEA 2023.1.1 Ultimate Edition (IU-231.8770.65)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
Related to https://github.com/intellij-rust/intellij-rust/pull/10415 

<img width="829" alt="image" src="https://user-images.githubusercontent.com/49211026/235637684-3830997e-072d-43a6-9447-765ff12699fe.png">

## Steps to reproduce
This code is correct on the stable rust toolchain.
```rust
#![cfg_attr(all(test, fuzzing), feature(no_coverage))]

pub fn add(left: usize, right: usize) -> usize {
    left + right
}

```
